### ISSUE TITLE
Unused `path_prefix` config argument for `pyroscope.scrape`

### ISSUE THREAD

**Issue by ptodev** [grafana/agent#6256]
### What's wrong?

The docs for `pyroscope.scrape` include a [path_prefix](https://grafana.com/docs/agent/v0.38/flow/reference/components/pyroscope.scrape/#profiling_config-block) argument, but the code for `pyroscope.scrape` doesn't seem to do anything with it.

We need to get a confirmation from the Pyroscope team that this config attribute is actually useful and that we should make it work. Otherwise we could just delete it.

### Steps to reproduce

No need to reproduce.

**Comment by cyriltovena** [comment]
I can confirm this can be useful when running a custom path prefix. Pretty much likes Prometheus scraping path.

**Comment by hainenber** [comment]
CMIIW but the `pyroscope.scrape` component seems to allow defining custom paths for each `pprof`endpoint

For example, [`profile.goroutine` block](https://grafana.com/docs/agent/v0.39/flow/reference/components/pyroscope.scrape/#profilegoroutine-block)

Me thinks we can remove it :D

**Comment by ptodev** [comment]
@hainenber that's true, but it might be inconvenient to override for all types of profiles.

**Comment by hainenber** [comment]
While it's tedious to do the override but I think that gives a more granular control to the users. In addition, they don't need to check other options that would override theirs. Just simple `path` for each profile :D

Overall, I think it's a good DX and bless to folks designing this initially 😄

**Comment by rfratto** [comment]
Hi there :wave: 

On April 9, 2024, Grafana Labs announced [Grafana Alloy](https://github.com/grafana/alloy), the spirital successor to Grafana Agent and the final form of Grafana Agent flow mode. As a result, Grafana Agent has been deprecated and will only be receiving bug and security fixes until its end-of-life around November 1, 2025. 

To make things easier for maintainers, we're in the process of migrating all issues tagged variant/flow to the Grafana Alloy repository to have a single home for tracking issues. This issue is likely something we'll want to address in both Grafana Alloy and Grafana Agent, so just because it's being moved doesn't mean we won't address the issue in Grafana Agent :)
